akka-remote/src/main/scala/akka/remote/artery/tcp/SSLEngineProvider.scala

```diff
@@ -73,119 +62,66 @@ class ConfigSSLEngineProvider(protected val config: Config, protected val log: M
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyStore: String = config.getString("key-store")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLTrustStore: String = config.getString("trust-store")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyStorePassword: String = config.getString("key-store-password")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLKeyPassword: String = config.getString("key-password")
+  @deprecated("Implement and setup a SslManagersProvider instead of using this setting.", "2.6.6")
   val SSLTrustStorePassword: String = config.getString("trust-store-password")
-  val SSLEnabledAlgorithms: Set[String] = immutableSeq(config.getStringList("enabled-algorithms")).to(Set)
-  val SSLProtocol: String = config.getString("protocol")
+
+  // Set up RNG for the SSLEngine
+  @deprecated("Use SecureRandomFactory directly.", "2.6.6")
   val SSLRandomNumberGenerator: String = config.getString("random-number-generator")
-  val SSLRequireMutualAuthentication: Boolean = config.getBoolean("require-mutual-authentication")
-  val HostnameVerification: Boolean = config.getBoolean("hostname-verification")
-
-  private lazy val sslContext: SSLContext = {
-    // log hostname verification warning once
-    if (HostnameVerification)
-      log.debug("TLS/SSL hostname verification is enabled.")
-    else
-      log.info(
-        LogMarker.Security,
-        "TLS/SSL hostname verification is disabled. See Akka reference documentation for more information.")
-
-    constructContext()
-  }
-
-  private def constructContext(): SSLContext = {
-    try {
-      val rng = createSecureRandom()
-      val ctx = SSLContext.getInstance(SSLProtocol)
-      ctx.init(keyManagers, trustManagers, rng)
-      ctx
-    } catch {
-      case e: FileNotFoundException =>
-        throw new SslTransportException(
-          "Server SSL connection could not be established because key store could not be loaded",
-          e)
-      case e: IOException =>
-        throw new SslTransportException("Server SSL connection could not be established because: " + e.getMessage, e)
-      case e: GeneralSecurityException =>
-        throw new SslTransportException(
-          "Server SSL connection could not be established because SSL context could not be constructed",
-          e)
-    }
-  }
+
+  // This exists as a field for backwards compat only (see usages below)
+  private val keyStoreProviders: JksManagerProviders = new JksManagerProviders(config, loadKeystore _)
+  private val rng: SecureRandom = SecureRandomFactory.createSecureRandom(SSLRandomNumberGenerator, log)
+  private lazy val sslFactory: SslFactory = new SslFactory(config, keyStoreProviders, rng)(log)
+  private lazy val sslContext: SSLContext = sslFactory.sslContext
```

Comment:
Implementing an `SSLEngineProvider` is reduced to these 4 lines which are used to implement the 4 methods in the `SSLEngineProvider` trait.